flocker/node/agents/test/test_blockdevice.py

```diff
@@ -5501,11 +5506,25 @@ def test_current_machine_is_live(self):
                           self.assertIn(self.api.compute_instance_id(), live))
             return d
 
+        def test_current_machine_has_appropriate_ip(self):
+            """
+            The machine's known IP is set for the current node.
+            """
+            local_addresses = set(unicode(i) for i in get_all_ips()
+                                  if i != b"127.0.0.1")
+            d = self.async_cloud_api.list_live_nodes()
+            d.addCallback(
+                lambda live:
+                self.assertTrue(
+                    set(live[self.api.compute_instance_id()]).intersection(
+                        local_addresses)))
+            return d
+
         def test_list_live_nodes(self):
             """
             ``list_live_nodes`` returns an iterable of unicode values.
             """
-            live_nodes = self.api.list_live_nodes()
+            live_nodes = list(self.api.list_live_nodes())
             self.assertThat(live_nodes, AllMatch(IsInstance(unicode)))
```

Comment:
Considering how brittle getting the wrong type may be, strongly prefer to verify the types of everything returned.

```
self.assertThat(live_nodes.itieritems(), AllMatch(
    MatchesListwise(
        (IsInstance(unicode), AllMatch(IsInstance(unicode)))
    )
))
```